docs/static/transforming-data.asciidoc

```diff
@@ -0,0 +1,520 @@
+[[transformation]]
+== Transforming Data
+
+With over 200 plugins in the Logstash plugin ecosystem, it's sometimes
+challenging to choose the best plugin to meet your data processing needs.
+In this section, we've collected a list of popular plugins and organized them
+according to their processing capabilities:
+
+* <<core-operations>>
+* <<data-deserialization>>
+* <<field-extraction>>
+* <<lookup-enrichment>>
+
+[[core-operations]]
+=== Performing Core Operations
+
+The plugins described in this section are useful for core operations, such as
+aggregating, mutating, and dropping events.
+
+Also see <<filter-plugins>> for the full list of available filter plugins. 
+
+<<plugins-filters-aggregate,aggregate filter>>::
+
+Aggregates information from several events (typically log lines) that belong to
+the same task and pushes the aggregated information into a final task event.
+For example:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  aggregate {
+    task_id => "%{country_name}"
+    code => "
+      map['country_name'] = event.get('country_name')
+      map['towns'] ||= []
+      map['towns'] << {'town_name' => event.get('town_name')}
+      event.cancel()
+    "
+    push_previous_map_as_event => true
+    timeout => 3
+  }
+}
+--------------------------------------------------------------------------------
+
+
+<<plugins-filters-date,date filter>>::
+
+Parses dates from fields to use as Logstash timestamps for events.
++
+The following config parses a field called `logdate` to set the Logstash
+timestamp:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  date {
```

Comment:
I feel this one is relatively self-explanatory and fine as-is. If they want more details, they can drill into the respective plugin page.